pkg/3scale/amp/component/redis.go

```diff
@@ -568,3 +490,118 @@ func (redis *Redis) buildEnv() []v1.EnvVar {
 }
 
 ////// End System Redis
+
+// //// Redis Config Map Begin
+type RedisConfigMap struct {
+	Options *RedisConfigMapOptions
+}
+
+func NewRedisConfigMap(options *RedisConfigMapOptions) *RedisConfigMap {
+	return &RedisConfigMap{Options: options}
+}
+
+func (r *RedisConfigMap) ConfigMap() *v1.ConfigMap {
+	return &v1.ConfigMap{
+		TypeMeta:   metav1.TypeMeta{Kind: "ConfigMap", APIVersion: "v1"},
+		ObjectMeta: r.buildConfigMapObjectMeta(),
+		Data:       r.buildConfigMapData(),
+	}
+}
+
+func (r *RedisConfigMap) buildConfigMapObjectMeta() metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:      redisConfigVolumeName,
+		Labels:    r.Options.Labels,
+		Namespace: r.Options.Namespace,
+	}
+}
+
+func (redis *Redis) buildConfigMapTypeMeta() metav1.TypeMeta {
+	return metav1.TypeMeta{
+		Kind:       "ConfigMap",
+		APIVersion: "v1",
+	}
+}
+
+func (r *RedisConfigMap) buildConfigMapData() map[string]string {
+	return map[string]string{
+		"redis.conf": r.getRedisConfData(),
+	}
+}
+
+func (r *RedisConfigMap) getRedisConfData() string { // TODO read this from a real file
```

Comment:
@valerymo are we happy with the fact that now, not only we disable the "replica/slave of" commands but also make it impossible to add any new configuration that's different to the below?
The configuration can't be extended or edited - is this a problem in your opinion?

Reply:
I get that but it seems like breaking current flow (that's been there for many years) - it pretty much means we are going to from now, revert everything (if there are changes present) to what we want. Let's discuss this further.

Reply:
Thank you for comment Michal. Yes, it's security restrictions,To prevent config map manipulation, and attempts to "play" with "small" changes in `rename-command` records in CM or in redis.conf itself. Sure, we can discuss at meeting. Thanks again

Reply:
It's for Internal Redis only. It's managed by Operator ..  Yes, will discuss it. Thank you

Reply:
discussed and updated, Thanks to @briangallagher  and @MStokluska